### PR TITLE
{AKS} Refactor a fourteenth part of the create sub-command in acs module

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

* **replace the implementation of `aks_create` with decorator**
* refactor auto_scaler_profile, sku, extended_location and custom headers
* finish `construct_default_mc_profile` and add `create_mc`
* clean up and add more test cases
* backward compatibility

Unit test result:

> "covered_lines": 1168,
> "num_statements": 1184,
> "percent_covered": 98.64864864864865,
> "missing_lines": 16,
> "excluded_lines": 0

Live test result:

* All passed.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # --------------------------------------------------------------------------------------------
 
 import re
+import time
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:7]: where's is this time used.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -2084,6 +2084,25 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
                no_wait=False,
                yes=False,
                enable_azure_rbac=False):
+    # decorator pattern
+    raw_parameters = locals()
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2088]: separate this into a different PR, so we can revert this easily.

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/decorator.py:7]: time is used to wait for a while before sending retry request

File: src/azure-cli/azure/cli/command_modules/acs/decorator.py [hunk 3]
```diff
@@ -3672,6 +3876,56 @@ def construct_default_mc(self) -> ManagedCluster:
...
+                    self.context.get_vnet_subnet_id(),
+                    self.context.get_enable_managed_identity(),
+                    self.context.get_attach_acr(),
+                    self.context.get_intermediate("custom_headers"),
+                    self.context.get_no_wait())
+                return created_cluster
+            except CloudError as ex:
+                retry_exception = ex
+                if 'not found in Active Directory tenant' in ex.message:
+                    time.sleep(3)
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/decorator.py:3928]: This is the place where time is used.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: Sure, changed in commit [#93a006f](https://github.com/Azure/azure-cli/pull/19583/commits/93a006f79172e8f3c5338e4b3cd05502f4cd361d).

**Review by andyliuliming — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]
